### ISSUE TITLE
Inconsistent dataframe output behaviour depending on the datatype of the last column

### ISSUE THREAD

**Issue by Khris777** [pandas-dev/pandas#21166]
After the update last friday printing a dataframe in ipython (Spyder) has become inconsistent:

I'll just create a test dataframe with 6 columns of different data types with a string column with longer entries:

	import pandas as pd
	test1 = ['AAAAAAAAAAAAAAA','BBBBBBBBBBBBBBBB','CCCCCCCCCCCCCCCCCCCCCCCCC','DDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD']
	test2 = [123,456,789,101]
	test3 = [5.234,9.5676,3.1415,78.66]
	l1 = test1
	l2 = test2
	l3 = test3
	l4 = test2
	l5 = test1
	l6 = test3
	df = pd.DataFrame({'C1':l1,'C2':l2,'C3':l3,'C4':l4,'C5':l5,'C6':l6})

This gives this output with 4 columns omitted:

												 C1   ...          C6
	0                           AAAAAAAAAAAAAAA   ...      5.2340
	1                          BBBBBBBBBBBBBBBB   ...      9.5676
	2                 CCCCCCCCCCCCCCCCCCCCCCCCC   ...      3.1415
	3  DDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD   ...     78.6600

On a sidenote: Apparently due to some changes somewhere `display.max_columns` now is set to 0 in Spyder, before it was always 20 due to missing auto-detection.

Now I'm changing every column to the string column:

	l1 = test1
	l2 = test1
	l3 = test1
	l4 = test1
	l5 = test1
	l6 = test1
	df = pd.DataFrame({'C1':l1,'C2':l2,'C3':l3,'C4':l4,'C5':l5,'C6':l6})

Output:

											 C1                                        C2                                        C3                                        C4                                        C5                                        C6
	0                           AAAAAAAAAAAAAAA                           AAAAAAAAAAAAAAA                           AAAAAAAAAAAAAAA                           AAAAAAAAAAAAAAA                           AAAAAAAAAAAAAAA                           AAAAAAAAAAAAAAA
	1                          BBBBBBBBBBBBBBBB                          BBBBBBBBBBBBBBBB                          BBBBBBBBBBBBBBBB                          BBBBBBBBBBBBBBBB                          BBBBBBBBBBBBBBBB                          BBBBBBBBBBBBBBBB
	2                 CCCCCCCCCCCCCCCCCCCCCCCCC                 CCCCCCCCCCCCCCCCCCCCCCCCC                 CCCCCCCCCCCCCCCCCCCCCCCCC                 CCCCCCCCCCCCCCCCCCCCCCCCC                 CCCCCCCCCCCCCCCCCCCCCCCCC                 CCCCCCCCCCCCCCCCCCCCCCCCC
	3  DDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD  DDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD  DDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD  DDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD  DDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD  DDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD

Now for some reason nothing is omitted, but while it looks somehow tidy with the formatting in here in the narrow ipython console it looks like an [awful unreadable mess](https://ibb.co/jpbE6T).

This only happens with the string column. If I only use a lot of float columns e.g. the behaviour is good.

EDIT: Apparently the cause is not that the whole dataframe is of string type, but the type of the last column. Take the first example and change only the last column to the string column:

	l1 = test1
	l2 = test2
	l3 = test3
	l4 = test2
	l5 = test1
	l6 = test1
	df = pd.DataFrame({'C1':l1,'C2':l2,'C3':l3,'C4':l4,'C5':l5,'C6':l6})

Output is suddenly not shortened anymore:

											 C1   C2       C3   C4                                        C5                                        C6
	0                           AAAAAAAAAAAAAAA  123   5.2340  123                           AAAAAAAAAAAAAAA                           AAAAAAAAAAAAAAA
	1                          BBBBBBBBBBBBBBBB  456   9.5676  456                          BBBBBBBBBBBBBBBB                          BBBBBBBBBBBBBBBB
	2                 CCCCCCCCCCCCCCCCCCCCCCCCC  789   3.1415  789                 CCCCCCCCCCCCCCCCCCCCCCCCC                 CCCCCCCCCCCCCCCCCCCCCCCCC
	3  DDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD  101  78.6600  101  DDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD  DDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.5.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 79 Stepping 1, GenuineIntel
byteorder: little
LC_ALL: None
LANG: en
LOCALE: None.None

pandas: 0.23.0
pytest: 3.5.1
pip: 10.0.1
setuptools: 39.0.1
Cython: 0.28.2
numpy: 1.14.3
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: 6.4.0
sphinx: 1.7.4
patsy: 0.5.0
dateutil: 2.7.2
pytz: 2018.4
blosc: None
bottleneck: 1.2.1
tables: 3.4.3
numexpr: 2.6.4
feather: None
matplotlib: 2.2.2
openpyxl: 2.5.3
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 1.0.4
lxml: 4.2.1
bs4: 4.6.0
html5lib: 1.0.1
sqlalchemy: 1.2.7
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: 0.1.5
pandas_gbq: None
pandas_datareader: None
</details>

**Comment by Khris777** [comment]
You are correct, it seems to be a Spyder-issue, I will post it over there then.

But now I stumbled over something else:

I opened the standard Command Prompt on Windows 10 and pasted the test code.

Now if I try to view the dataframe I don't see anything at all, it looks like this now:

	>>> print(df)
	  ...
	0 ...
	1 ...
	2 ...
	3 ...

	[4 rows x 6 columns]

No idea if that's just because the Windows Terminal is just as bad as it is or something else.

**Comment by mroeschke** [comment]
This looks to be fixed. Will be available in the upcoming release (0.24). Closing

```
In [39]: print(df)
                                         C1   C2       C3   C4                                        C5       C6
0                           AAAAAAAAAAAAAAA  123   5.2340  123                           AAAAAAAAAAAAAAA   5.2340
1                          BBBBBBBBBBBBBBBB  456   9.5676  456                          BBBBBBBBBBBBBBBB   9.5676
2                 CCCCCCCCCCCCCCCCCCCCCCCCC  789   3.1415  789                 CCCCCCCCCCCCCCCCCCCCCCCCC   3.1415
3  DDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD  101  78.6600  101  DDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD  78.6600
```
